qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -200,7 +199,8 @@ def _discover_backends_from_instance(self, instance: str) -> List[str]:
                     new_client = self._create_new_cloud_api_client(instance)
                     self._api_clients.update({instance: new_client})
                     self._active_api_client = new_client
-            return self._active_api_client.list_backends()
+            self._backends_list = self._active_api_client.list_backends()
+            return [backend["name"] for backend in self._backends_list]
         # On staging there some invalid instances returned that 403 when retrieving backends
```

Comment:
We still have to store the `/backends` API response somewhere in the service object. I think it's best to handle that here in `_discover_backends_from_instance()`?